dev/io.openliberty.mcp.internal_fat/fat/src/io/openliberty/mcp/internal/fat/tool/DeploymentProblemTest.java

```diff
@@ -0,0 +1,77 @@
+/*******************************************************************************
+ * Copyright (c) 2025 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.mcp.internal.fat.tool;
+
+import static com.ibm.websphere.simplicity.ShrinkHelper.DeployOptions.DISABLE_VALIDATION;
+import static com.ibm.websphere.simplicity.ShrinkHelper.DeployOptions.SERVER_ONLY;
+import static org.junit.Assert.assertTrue;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.FATServletClient;
+import io.openliberty.mcp.internal.fat.tool.duplicateToolErrorTestApp.DuplicateToolErrorTest;
+import io.openliberty.mcp.internal.fat.utils.HttpTestUtils;
+
+/**
+ *
+ */
+@RunWith(FATRunner.class)
+public class DeploymentProblemTest extends FATServletClient {
+
+    @Server("mcp-server")
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void setup() throws Exception {
+        WebArchive war = ShrinkWrap.create(WebArchive.class, "DeploymentProblemTest.war").addPackage(DuplicateToolErrorTest.class.getPackage());
+        ShrinkHelper.exportDropinAppToServer(server, war, DISABLE_VALIDATION, SERVER_ONLY);
+
+        server.startServer();
+    }
+
+    @AfterClass
+    public static void teardown() throws Exception {
+        server.stopServer("CWWKZ0002E");
+    }
+
+    @Test
+    public void testDuplicateToolDeploymentError() throws Exception {
+        String request = """
+                        {
+                          "jsonrpc": "2.0",
+                          "id": 1,
+                          "method": "tools/list",
+                          "params": {
+                            "cursor": "optional-cursor-value"
+                          }
+                        }
+                        """;
+
+        boolean deploymentErrorOccured = false;
+
+        try {
+            HttpTestUtils.callMCP(server, "/toolTest", request);
+        } catch (Exception e) {
+            deploymentErrorOccured = true;
+        }
+
+        assertTrue("Expected a deployment error due to duplicate MCP Tools being present, no error was present", deploymentErrorOccured);
+    }
```

Comment:
This is not the right way to test this. We should be checking the log to ensure:
1. the app did not start (we should get a "your app failed to start" error
2. a useful error message is logged

use `LibertyServer.waitForStringInLog` and `LibertyServer.findStringsInLog` to do this.